articles/architecture-scenarios/b2c/tenant-architecture.md

```diff
@@ -0,0 +1,83 @@
+
+::: warning
+Tenant names cannot be changed or reused (even if the original tenant has been deleted). Make sure that you are happy with your names before you set up the tenants.
+:::
+
+## Design considerations
+
+When you begin the process of designing your tenant architecture, consider the following questions:
+
+1. What should the URL be when the Authorization Server (Auth0) presents a page to the user? You tenant name is part of the URL, and as part of Phase 1, you may be setting up [custom domains](#custom-domains).
+
+1. How will you support the Software Development Life Cycle? Do you have development, QA, and production environments, with each needing its own tenant? Do you have an additional staging environment, or are you working with a minimalist setup featuring just development and production environments?
+
+1. Are there any other projects that might impact your Auth0 integration? For example, if there are future projects in the pipeline, consider your setup carefully to minimize the likelihood that you'll have to migrate your existing tenant to another at that time.
+
+1. How do you make sure that all of your tenants are appropriately associated with your Auth0 contract? Auth0 has a [set of procedures](#associating-tenants) for you to follow if you want all of your tenants associated with each other.
+
+## Custom domains
+
+When you set up your tenant, the URL to access that tenant will be `https://YOUR_TENANT_NAME.auth0.com` where `YOUR_TENANT_NAME` is the name of your tenant. 
+
+::: note
+You can create one custom domain for each tenant. If you need to use more than one domain name, please see [Customization](#).
+:::
+
+However, you can implement a [custom domain](/custom-domains) for your tenant (which is the authorization server). This is more than a branding task -- it also provides you with security benefits:
+
+1. You'll find it easier to facilitate communications within an iFrame, which some browsers [make difficult to do](/api-auth/token-renewal-in-safari) if you have a shared domain name
+
+1. It's [hard for malicious parties to phish](https://auth0.com/blog/introducing-custom-domains-preview-with-auth0/) using your domain with a custom URL. If you have a custom URL, the phisher must also create a custom URL to match yours.
+
+1. It will be easier for your users to recognize suspicious URLs whenever they need to enter passwords.
+
+Create a custom domain in all of your environments early on in the implementation process to make sure that you are testing consistently between the environments. Your custom domain should give your users confidence that the site they're on is the one where they should be entering their credentials.
+
+### Best practice for custom domains
```

Comment:
Why does this have a section style as opposed to a call-out style for best practice? Consistency of style within this set of documentation would be preferable I feel.